added ENS resolver hooks, display results in the console   (Ready for review)

Seroxdesign opened the pull request:
## Progress

- [x]  Added two functions in the hooks folder. one resolves ENS to address and one resolves address to ENS

### Remaining work

What's left:

- [x]  Adding checks for search terms to ensure we call the right method
- [x]  Adding a debounce to minimise spam to RPC
- [x]  Showing results under search bar and on the account view page

Seroxdesign commented:
# THIS PR IS A WORK IN PROGRESS

kasparkallas commented:
Glad to see interest in contributing to the Console!

Some comments on how to solve this story:

* We use [RTK-Query](https://redux-toolkit.js.org/rtk-query/overview) to handle cache'ing and some request debounce'ing. Ethers' `JsonRpcBatchProvider` & `FallbackProvider` are used for further throttling & robustness.
* We implemented a solution for the upcoming Dashboard and would like to re-use it in the Console. If the solution is not good then we'd like to know why. I put the snippets of the code in another ENS pull request: https://github.com/superfluid-finance/superfluid-console/pull/88#issuecomment-1159093336 Effectively this bounty can be grabbed by just following these snippets.

Seroxdesign commented:
Still reading through all the snippets, and deciding how to fit them in. Expecting results by July 9th.

Seroxdesign commented:
@kasparkallas @vmichalik @sunnyjaycer 

Ready for review

kasparkallas commented on src/components/AddressBook.tsx:
```diff
@@ -23,18 +24,24 @@ import {
 export const AddressBookButton: FC<{
   network: Network;
   address: string;
+  description: any;
```
What's a "description". Why type "any"?

kasparkallas commented on src/redux/slices/addressBook.slice.ts:
```diff
@@ -5,6 +5,7 @@ import {
 import { Network } from '../networks';
 import {RootState} from "../store";
 import {REHYDRATE} from "redux-persist";
+import { add } from 'lodash';
```
Not used?

Seroxdesign replied:
Removed

kasparkallas commented on src/redux/store.ts:
```diff
@@ -61,7 +62,9 @@ export const makeStore = wrapMakeStore(() => {
       [rpcApi.reducerPath]: rpcApi.reducer,
       [sfSubgraph.reducerPath]: sfSubgraph.reducer,
       [themePreferenceSlice.name]: themePreferenceSlice.reducer,
-      [addressBookSlice.name]: addressBookReducer
+      [addressBookSlice.name]: addressBookReducer,
+      [addressBookSlice.name]: addressBookReducer,
```
Double reducer

kasparkallas commented on package.json:
```diff
@@ -33,6 +33,7 @@
     "next-redux-cookie-wrapper": "^2.1.2",
     "next-redux-wrapper": "^7.0.5",
     "react": "^18.0.0",
+    "react-blockies": "^1.4.1",
```
Didn't see blockies anywhere

Seroxdesign replied:
Removed unused package and types

kasparkallas reviewed (commented):
Did a quick tour and seems to work overall! :+1:

Couple of comments:
* doesn't show spinner when still fetching ENS
* when on an account page and then going to another account's page then previous avatar is shownfor a short period of time (use RTK-query's `.currentData` over `.data` or set a _key_ for the page [use address as key for example])
* when both ENS and address book name are specified then the behaviour is weird with both showing and the address not showing
* UI is a bit messy and doesn't follow https://mui.com/material-ui/

Seroxdesign commented on src/pages/[_network]/accounts/[_id].tsx:
```diff
@@ -64,12 +65,15 @@ import ellipsisAddress from "../../../utils/ellipsisAddress";
 
+  const ensAddressQuery = ensApi.useLookupAddressQuery(address);
+
+  const ensName = ensAddressQuery.data?.name
```
Done

Seroxdesign commented:
I believe most problems have been resolved, although I might need to figure out how to display addressBook name and ENS name

Seroxdesign commented:
> Did a quick tour and seems to work overall! 👍
> 
> Couple of comments:
> 
>     * doesn't show spinner when still fetching ENS
> 
>     * when on an account page and then going to another account's page then previous avatar is shownfor a short period of time (use RTK-query's `.currentData` over `.data` or set a _key_ for the page [use address as key for example])
> 
>     * when both ENS and address book name are specified then the behavior is weird with both showing and the address not showing
> 
>     * UI is a bit messy and doesn't follow https://mui.com/material-ui/

- Switched to using currentData

- Added conditionals to avoid rendering both address book name and ens name at the same time

- Slightly improved UI on account page

- Spinner isn't loading immediately because it takes a second to resolve ENS name to address for search

@kasparkallas Let me know if anything else needs work

Seroxdesign commented:
So seems like there's an issue I am not handling right, I don't know what the best way to check name before I resolve to address

vmichalik reviewed (approved):
looks good to me!

elvijsTDL commented:
Doesn't work on Brave and Firefox browsers
https://user-images.githubusercontent.com/77115130/179575974-d69c8ae3-4162-4780-9939-70d311077a74.mov

The non-ens address in at the top of the page is alligned a little bit off now and it is smaller, here it is now
<img width="639" alt="Screenshot 2022-07-18 at 21 20 08" src="https://user-images.githubusercontent.com/77115130/179576933-5e2f4c73-09b9-4517-8d33-46673f5752f1.png">

And here is the before
<img width="639" alt="Screenshot 2022-07-18 at 21 21 37" src="https://user-images.githubusercontent.com/77115130/179577147-7091aeff-75d1-4990-9c86-9535c7e0ff9e.png">


@vmichalik Should we also include showing the ENS names in the tables and using them in filters in this ticket?

elvijsTDL reviewed (changes requested):
Was on a slower network and the loading spinner doesn't show up when the ENS lookup is happening, actually thought the ENS lookup didn't work on Brave and Firefox because of that, could we please have one while the lookup is still happening

Seroxdesign commented:
> The non-ens address in at the top of the page is alligned a little bit off now and it is smaller, here it is now <img alt="Screenshot 2022-07-18 at 21 20 08" width="639" src="https://user-images.githubusercontent.com/77115130/179576933-5e2f4c73-09b9-4517-8d33-46673f5752f1.png">
> 
> And here is the before <img alt="Screenshot 2022-07-18 at 21 21 37" width="639" src="https://user-images.githubusercontent.com/77115130/179577147-7091aeff-75d1-4990-9c86-9535c7e0ff9e.png">
> 
> @vmichalik Should we also include showing the ENS names in the tables and using them in filters in this ticket?

I changed the size of the text to comply with the ENS frontend standard, 

![image](https://user-images.githubusercontent.com/69639595/179968652-89c93de9-84fd-4030-af94-926d82c9c1ed.png)

An example of how this is relevant, I wanted the ENS to be bigger than the address, although this can be changed back to the original style

Seroxdesign commented:
> Was on a slower network and the loading spinner doesn't show up when the ENS lookup is happening, actually thought the ENS lookup didn't work on Brave and Firefox because of that, could we please have one while the lookup is still happening , here is a video:
> Screen.Recording.2022-07-18.at.21.40.42.mov
> 
> Also address book entries now show the input string too instead of just the result: <img alt="Screenshot 2022-07-18 at 22 00 44" width="1275" src="https://user-images.githubusercontent.com/77115130/179583989-88ab6e8c-509a-4879-afa7-fcab30a1b734.png">

The text is a simple fix, I will remove showing the text on top of the address and instead only showed related addresses on ENS search.

As for the spinner, I tried fixing this yesterday but I am still trying to figure out the best way to do this without changing how the spinner renders. I will try again today and if I can't find a solution I will request support

Seroxdesign commented:
Before:

![image](https://user-images.githubusercontent.com/69639595/179981873-e19873a3-f811-43ce-aa89-52539a545a93.png)
![image](https://user-images.githubusercontent.com/69639595/179981978-b495fccb-7e7b-4932-9b64-d4b9388cf843.png)


After:

![image](https://user-images.githubusercontent.com/69639595/179982013-02cc23af-dac2-42c8-923e-cebca2899580.png)
![image](https://user-images.githubusercontent.com/69639595/179982127-1e4904c9-965e-4730-b596-66a00d100f66.png)

What's left:

- []  Figuring out how to show spinner correctly.

Seroxdesign commented:
@elvijsTDL @kasparkallas I am having trouble with the spinner task, I would appreicate any pointer or advice,

Seroxdesign commented:
> Was on a slower network and the loading spinner doesn't show up when the ENS lookup is happening, actually thought the ENS lookup didn't work on Brave and Firefox because of that, could we please have one while the lookup is still happening , here is a video:
> Screen.Recording.2022-07-18.at.21.40.42.mov
> 
> Also address book entries now show the input string too instead of just the result: <img alt="Screenshot 2022-07-18 at 22 00 44" width="1275" src="https://user-images.githubusercontent.com/77115130/179583989-88ab6e8c-509a-4879-afa7-fcab30a1b734.png">

![image](https://user-images.githubusercontent.com/69639595/181030423-9e30a7e8-aaa0-4847-afe0-da5d63d55fe4.png)

Fixed using isFetching on ensQuery

vmichalik commented:
@Seroxdesign, we really appreciate your efforts- you took to the task quickly, were responsive to feedback and suggestions and kept making tweaks to improve, so I'm more than happy to see this bounty paid out. 

With Dashboard V2 still under such active development we likely won't have time to do a Console release containing this feature for a few weeks. However, we hope you enjoy the bounty payout, learnt more about ENS and the SDK-Redux and stick around for future bounties too🥇 

Will DM you on Discord for next steps

elvijsTDL commented:
LGTM  👍

kasparkallas reviewed (approved):
When there's an address book entry then ENS name is not displayed anywhere. Not sure if I like it... The ENS name should still be visible somewhere but let's handle that in the next iteration.
